frontend/lib/src/components/shared/Modal/Modal.tsx

```diff
@@ -135,8 +136,41 @@ const ModalButton: FunctionComponent<
   </StyledModalButton>
 )
 
-function Modal(props: ModalProps): ReactElement {
-  const { spacing, radii, colors }: EmotionTheme = useTheme()
+export type StreamlitModalProps = Omit<ModalProps, "size"> & {
+  size?: "auto" | "default" | "full"
+}
+
+/**
+ * Maps our own StreamlitModal size to the Baseweb Modal size or a calculated string.
+ * This abstraction allows us later to swap the Baseweb Modal size without touching
+ * the other components again.
+ *
+ * @param size the StreamlitModal size to be mapped
+ * @param width the width of the modal if 'full' size is selected
+ * @param padding the padding added to the modal if 'full' size is selected
+ * @returns the Baseweb Modal comaptible size
+ */
+export function calculateModalSize(
```

Comment:
note: I moved this function from the `Dialog.tsx` component so that the `DeployDialog` can re-use it